tests/cross-account-associations/main.tf

```diff
@@ -0,0 +1,195 @@
+/*
+  The workflow for cross-account associations is a bit of a dance. There are several things that must
+  take place in the "owner" account, in order for the associations to work properly in the "member"
+  account.
+
+  Owner account:
+  - Create the resolver
+  - Create the resolver rule and/or the query log config
+  - Create the RAM share
+  - Associate the resolver rule and query log config resources to the RAM share
+  - Invite the "member" account to the RAM share as a principal
+
+  Member account:
+  - Accept the invitation to the RAM share
+  - Create the associations for the resolver rule and/or the query log config
+
+  IMPORTANT: The associations in the member account are dependent on *several* resources that are not
+  direct attributes of the association. In order for terraform to properly manage the ordering of create/destroy
+  actions, use `depends_on` for the RAM share accepter, the principal association, and the resource association.
+*/
+
+provider "aws" {
+  region  = "us-east-1"
+  profile = "aws"
```

Comment:
Shouln't the test provider profiles and regions be variables?

Reply:
We could, though it may be a little harder to use in the wrapper, in automation. Which region for which provider for which test? Does each test have uniquely named vars so it can be passed explicitly per test per provider?

Using a hardcoded profile name is a bit of a test of its own... The terraform test suite already does this, using exactly these profile names. So thought we might follow their lead here...